lib/base/request.js

```diff
@@ -40,6 +40,18 @@ class Request extends EventEmitter {
     this.parameters = {}
   }
 
+  /**
+   * Generate sql string and set imput parameters from tagged template string.
+   *
+   * @param {Template literal} template
+   * @return {String}
+   */
+  template () {
```

Comment:
I'm confused about how this is accepting a callback... can you explain?

Reply:
template() doesn't accept a callback, it just generates the sql from the template and sets the input parameters. It's more of a replacement for calling input() and writing an sql string with @params.

It allows you to use the callback version of request.query with template literals. (see usage above: 
```request.query(request.template`select * from mytable where id = ${value}`, (err, result) =>...``` 
)

It's to work around the fact that the template literal version of request.query() doesn't accept a callback. I don't think it would be possible to pass it a callback... unless you would pass the callback in the template or something but this might be weird?